src/WarcraftLegacies.Source/Quests/Sentinels/QuestFeathermoon.cs

```diff
@@ -1,42 +1,61 @@
-﻿using MacroTools.FactionSystem;
+﻿using System.Collections.Generic;
+using MacroTools.Commands;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
 using MacroTools.LegendSystem;
 using MacroTools.ObjectiveSystem.Objectives.ControlPointBased;
 using MacroTools.ObjectiveSystem.Objectives.LegendBased;
+using MacroTools.ObjectiveSystem.Objectives.UnitBased;
 using MacroTools.QuestSystem;
+using WCSharp.Shared.Data;
 
 namespace WarcraftLegacies.Source.Quests.Sentinels
 {
-  /// <summary>
-  /// Unlocks Feathermoon Stronghold upon securing the area and bringing a hero to it.
-  /// </summary>
   public sealed class QuestFeathermoon : QuestData
   {
-    /// <summary>
-    /// Initializes a new instance of <see cref="QuestFeathermoon"/>.
-    /// </summary>
-    public QuestFeathermoon(Capital feathermoon) : base("Shores of Feathermoon",
-      "Feathermoon Stronghold will undoublty fall to the assault of the Old gods, we will need to recapture it. ",
+    private readonly List<unit> _rescueUnits;
+    private readonly Capital _feathermoon; // Reference to the capital Feathermoon unit
+
+    public QuestFeathermoon(Rectangle questRect, Capital feathermoon) : base("Shores of Feathermoon",
+      "Feathermoon Stronghold will undoubtedly fall to the assault of the Old gods, we will need to restore it.",
       @"ReplaceableTextures\CommandButtons\BTNBearDen.blp")
     {
-      AddObjective(new ObjectiveControlCapital(feathermoon, false));
-      AddObjective(new ObjectiveControlPoint(UNIT_N05U_FEATHERMOON_STRONGHOLD));
+     
+      _feathermoon = feathermoon;
+      AddObjective(new ObjectiveBuildUniqueBuildingsInRect(questRect, "in Feathermoon", 3));
+      AddObjective(new ObjectiveControlPoint(UNIT_N05U_FEATHERMOON_STRONGHOLD, 900));
       ResearchId = UPGRADE_R06M_QUEST_COMPLETED_SHORES_OF_FEATHERMOON;
-    }
 
-    /// <inheritdoc />
-    public override string RewardFlavour =>
-      "Feathermoon Stronghold has been recaptured and has joined the Sentinels in their war effort";
+      // Prepare units for rescue excluding the capital
+      _rescueUnits = Regions.FeathermoonUnlock.PrepareUnitsForRescue(RescuePreparationMode.HideNonStructures, unit => unit != feathermoon.Unit);
+    }
 
     /// <inheritdoc />
-    protected override string RewardDescription => 
-      $"Learn to train Naisha from the {GetObjectName(UNIT_E00R_ALTAR_OF_WATCHERS_SENTINEL_ALTAR)}";
+    protected override void OnComplete(Faction completingFaction)
+    {
+      // Include the capital unit in the rescue group
```

Comment:
How come this is done this way? Intuitively I'd think if the capital is being rescued it should be prepared for rescue.